src/modules/swap/ModuleSwap.vue

```diff
@@ -1167,6 +1166,16 @@ export default {
       this.localGasPrice = e.gasPrice;
       if (this.currentTrade) this.currentTrade.gasPrice = this.localGasPrice;
       this.localGasType = e.gasType;
+    },
+    setTokenFromURL() {
+      if (Object.keys(this.$route.query).length > 0) {
+        const { fromToken, toToken, amount } = this.$route.query;
+        this.defaults = {
+          fromToken,
+          toToken
+        };
+        this.tokenInValue = `${amount}`;
```

Comment:
Can you add a check on whether amount exists? in case that it's `undefined`. Also, can you sanitize the query values with the `xss` library

Reply:
fixed